### PR TITLE
DEVPROD-8420 Fetch app id instead of boolean 

### PR DESCRIPTION
**Pull request by malikchaya2**
DEVPROD-8420

### Description
This supports a design update to display the app id instead of a boolean. 

### Testing
Updated tests

### CODE REVIEW

File: model/github_app_auth.go [hunk 1]
```diff
@@ -21,26 +21,30 @@ const (
 // FindOneGithubAppAuth finds the github app auth for the given project id
 func FindOneGithubAppAuth(projectId string) (*evergreen.GithubAppAuth, error) {
 	githubAppAuth := &evergreen.GithubAppAuth{}
-	q := db.Query(bson.M{githubAppAuthIdKey: projectId})
-	err := db.FindOneQ(GitHubAppAuthCollection, q, githubAppAuth)
+	err := db.FindOneQ(GitHubAppAuthCollection, byAppAuthID(projectId), githubAppAuth)
 	if adb.ResultsNotFound(err) {
 		return nil, nil
 	}
-	if err != nil {
```
Comment (ZackarySantana) [inline model/github_app_auth.go:29]: Nice refactor!

File: model/github_app_auth.go [hunk 2]
```diff
@@ -21,26 +21,30 @@ const (
...
 
-// HasGithubAppAuth checks if the github app auth for the given project id exists
-func HasGithubAppAuth(projectId string) (bool, error) {
-	var app *evergreen.GithubAppAuth
-	var err error
-	if app, err = FindOneGithubAppAuth(projectId); err != nil {
-		return false, err
+// byAppAuthID returns a query that finds a github app auth by the given identifier
+// corresponding to the project id
+func byAppAuthID(projectId string) db.Q {
```
Comment (ZackarySantana) [inline model/github_app_auth.go:33]: just a non-blocking nit, but should we call this `byGithubAppAuthID`? It works better for intellisense and keeps the naming conventions the same in the file. If you think not, you can just resolve this!

File: model/github_app_auth.go [hunk 3]
```diff
@@ -21,26 +21,30 @@ const (
...
+	return db.Query(bson.M{githubAppAuthAppIdKey: projectId})
+}
+
+// GetGitHubAppID returns the app id for the given project id
+func GetGitHubAppID(projectId string) (*int64, error) {
+	githubAppAuth := &evergreen.GithubAppAuth{}
+
+	q := byAppAuthID(projectId).WithFields(githubAppAuthIdKey)
```
Comment (ZackarySantana) [inline model/github_app_auth.go:41]: TIL `WithFields` applies a projection!

File: model/github_app_auth_test.go [hunk 4]
```diff
@@ -32,6 +32,26 @@ func TestFindOneGithubAppAuth(t *testing.T) {
...
+		Id:         "mongodb",
+		AppID:      1234,
+		PrivateKey: key,
+	}
+	err := UpsertGithubAppAuth(&githubAppAuth)
+	require.NoError(t, err)
+
+	appIDFromDB, err := GetGitHubAppID("mongodb")
+	require.NoError(t, err)
+	assert.Equal("mongodb", appIDFromDB)
```
Comment (ZackarySantana) [inline model/github_app_auth_test.go:52]: should this be 1234 (as an int)?

File: model/project_ref.go [hunk 5]
```diff
@@ -1983,18 +1983,21 @@ func GetProjectSettings(p *ProjectRef) (*ProjectSettings, error) {
...
 		GithubHooksEnabled: hasEvergreenAppInstalled,
 		Vars:               *projectVars,
 		Aliases:            projectAliases,
 		Subscriptions:      subscriptions,
-		HasGitHubApp:       hasGithubAppAuth,
+	}
+
+	githubAppID, err := GetGitHubAppID(p.Id)
+	if err != nil {
+		return nil, errors.Wrapf(err, "finding hadGithubApp for project '%s'", p.Id)
```
Comment (ZackarySantana) [inline model/project_ref.go:1996]: `hadGithubApp` -> userfriendly `github app` or new method `GetGitHubAppID`

**Review by ZackarySantana — APPROVED** [review]
LGTM mod the comments!

File: model/github_app_auth.go (hunk 1 above)
Comment (malikchaya2) [reply model/github_app_auth.go:29]: thank you :)

File: model/github_app_auth_test.go (hunk 4 above)
Comment (malikchaya2) [reply model/github_app_auth_test.go:52]: yes 🙈 
thank you!

File: model/github_app_auth.go (hunk 2 above)
Comment (malikchaya2) [reply model/github_app_auth.go:33]: yes, renamed it

File: model/project_ref.go (hunk 5 above)
Comment (malikchaya2) [reply model/project_ref.go]: true, thank you! fixed

**Pull request merged**